### ISSUE TITLE
Viewing annotated PDF documents

### ISSUE THREAD

**Issue by tillahoffmann** [philips/supernote-obsidian-plugin#31]
Thanks for the great plugin! Is it possible to view annotations for PDF documents stored in `.pdf.mark` files?

**Comment by philips** [comment]
No, it does not do annotations of PDFs. I would need to figure out a PDF parser, I think, which would be a whole engineering undertaking.

**Comment by philips** [comment]
I am going to close this for now as I have no intention of working on this at the moment. I would consider a design discsusion if someone else is interested in doing it.

**Comment by MijnheerD** [comment]
Hi @philips ! I am also interested in this feature and would like to try my hand implementing it :) I have coding experience, but not with JS unfortunately and I am also new to Obsidian (I actually moved to Obsidian because of your plugin). So it will be an interesting challenge.

I found [this repo](https://github.com/nelinory/SupernoteSharp) which implements reading in `.pdf.mark` files and converting to either PDF or PNG format. Based on this, my idea would be something like this:
- When opening a PDF in Obsidian, check whether a matching `.mark` is present. If so, offer the option to combine them.
- Convert that annotations file to PDF and somehow merge it with the original one.
- Save the merged PDF.

In a first stage, I would let Obsidian's native PDF viewer do the heavy lifting when it comes to displaying the result. We could recommend plugins like [PDF++](https://github.com/RyotaUshio/obsidian-pdf-plus) to enhance the reading experience, but I don't think we should implement PDF viewing inside this plugin.

If this all works, I think it would also be nice to have a way of processing "digest" files. I have been looking for a while on how to share these with other people (which do not have a Supernote) and have not found a satisfying solution yet.

I am curious to hear your thoughts!
